File: composition/cpp/Lune.cpp
#include "Lune.h"

//composition dans le constructeur directement ! (déclaration et instanciation dans le constructeur)
Lune::Lune(string nom, int taille, string nomBat, int production):nom(nom),taille(taille),batiment(nomBat,production){};
//destructeur
Lune::~Lune(){cout<<"destruction de la lune"<<endl;}

File: composition/java/Lune.java
package packCompo;

public class Lune {
    private String nom;
    private int taille;
    private Batiment batiment;

    //Constructeur par initialisation
    public Lune(String nom, int taille, String nomBat, int production){
        this.nom=nom;
        this.taille=taille;
        //composition comme ceci
        this.batiment=new Batiment(nomBat, production);
    }

    //destructeur
    public void finalize(){
        System.out.println("Lune détruite");
    }

    //méthode pour afficher le batiment
    public void affichage(){
        System.out.println("Batiment de la lune : "+batiment.getNom()+" production : "+batiment.getProduction());
    }

    //pas besoin de setter et de getter pour notre exemple
}